staging_upload.py

```diff
@@ -36,9 +45,133 @@ def die(msg):
     sys.exit(1)
 
 
-async def download():
-    pass
-    # windows / mac download from archive; linux download from task
+async def download(context):
+    abs_path_per_platform = {}
+    for platform, mar_url in LATEST_MAR_LOCATIONS.items():
+        platform_sub_folder = get_platform_base_folder(context, platform)
+
+        mar_abs_path = os.path.join(platform_sub_folder, get_sub_path_from_url(mar_url))
+        await download_file(context, url=mar_url, abs_filename=mar_abs_path)
```

Comment:
This flow will result in synchronous downloading, which is probably fine with 5 mars to download.
If we want to speed things up, we can do something like

```python
tasks = []
tasks.append(
    asyncio.ensure_future(
        download_file(context, url=mar_url, abs_filename=mar_abs_path)
    )
)
await raise_future_exceptions(tasks)
```

where `raise_future_exceptions` is https://github.com/mozilla-releng/scriptworker/blob/master/scriptworker/utils.py#L320-L327